src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -35,10 +37,17 @@ def _api_request(self, url_suffix, params=None):
     def list(self, **params):
         return self._api_request('/alerts', params)
 
-    @RESTController.Collection('POST')
-    def get_notifications_since(self, **last_notification):
+
+@ApiController('/prometheus/notifications', Scope.PROMETHEUS)
+class PrometheusNotifications(RESTController):
+
+    def list(self):
+        return PrometheusReceiver.notifications[-1:]
+
+    def get(self, n_id):
         notifications = PrometheusReceiver.notifications
-        if last_notification not in notifications:
-            return notifications[-1:]
-        index = notifications.index(last_notification)
+        index = -2
+        for i, notification in enumerate(notifications):
+            if notification['id'] == n_id:
+                index = i
```

Comment:
May it's just me but I find this counter-intuitive. `GET /<resource>/<id>` in RESTful APIs is expected to retrieve the single resource matching that `id`, not everything else. For a range of notifications, the usual practice in RESTful APIs would be: `GET /<resource>?from=<id_0>&to=<id_1>`.

Query params are automatically parsed and provided by Cherrypy, and `HttpParams` is the builder counterpart in the Angular side.

Reply:
Hm... but this is not a usual case that's why I used post before as the notifications from the Alertmanager are received by the backend without any id. They are also only stored there in an array which vanishes on reload.

As soon as a new notification (not the first notification) is received by the dashboard it will notify the user via toasty. That said, the frontend has no idea when a new notification is retrieved by the backend and it does not know how many notifications are stored.

But I will try to implement it instead with `?from=index`

Reply:
Interesting, so this seems a good candidate for a websocket? If I'm not wrong, using the query params should be just a couple of lines (plus fixing unit tests).

Reply:
Done